yt/frontends/enzo/io.py

```diff
@@ -75,16 +78,24 @@ def _read_particle_fields(self, chunks, ptf, selector):
                     continue
                 ds = f.get("/Grid%08i" % g.id)
                 for ptype, field_list in sorted(ptf.items()):
-                    if ptype != "io":
+                    if ptype == "io":
+                        if g.NumberOfParticles == 0:
```

Comment:
I think should this be `g.NumberOfParticles[ptype]`?

Reply:
oh... hmm. Looks like I was wrong about this? doesn't look like any tests failed due to this.

Reply:
oh, ya, I skimmed through too quickly. `NumberOfParticles` is not the same as `NumberOfActiveParticles`... Ignore above comments :)